app/grandchallenge/cases/serializers.py

```diff
@@ -50,3 +56,28 @@ class Meta:
             "algorithm_image",
             "api_url",
         ]
+
+
+class RawImageFileSerializer(serializers.ModelSerializer):
+    upload_session = HyperlinkedRelatedField(
+        queryset=RawImageUploadSession.objects.all(),
+        view_name="api:upload-session-detail",
+    )
+
+    def validate(self, attrs):
+        upload_session = attrs["upload_session"]
+        algorithm_image = upload_session.algorithm_image
+        user = self.context.get("request").user
+        if upload_session.creator != user:
+            raise ValidationError(
+                f"User {user} does not have permission to see this raw image upload session "
+            )
+        if algorithm_image.creator != user:
+            raise ValidationError(
+                f"User {user} does not have permission to see the corresponding algorithm image "
+            )
```

Comment:
These are not the correct checks, see the permissions tests for the algorithms `execution-session-create` view. 

We should check if the user has permission to view the upload session (this was set in `cases/models.py`) using `user.has_perm()`.

The algorithm permission check should be done in the upload session serializer, and should check if the user has the permission to view the algorithm that the algorithm image belongs to.